src/HDF5.jl

```diff
@@ -2,34 +2,39 @@ module HDF5
 
 using Base: unsafe_convert, StringVector
 using Requires: @require
+# needed for filter(f, tuple) in julia 1.3
+using Compat
 
 import Libdl
 import Mmap
 
-# needed for filter(f, tuple) in julia 1.3
-using Compat
-
 export
 @read, @write,
-h5open, h5read, h5rewrite, h5writeattr, h5readattr, h5write,
-a_create, a_delete, a_open, a_read, a_write, attrs,
-d_create, d_create_external, d_open, d_read, d_write,
-g_create, g_open,
-o_copy, o_delete, o_open,
-t_create, t_commit,
-p_create, get_access_properties, get_create_properties,
-dataspace, datatype,
-file, filename, name,
-get_chunk, get_datasets,
-iscontiguous, ishdf5, ismmappable,
-root, readmmap, set_dims!
-# Note: Public API requires module scoping
-### Public Functions:
-# g_info, o_info
-### Types:
+h5open, h5read, h5write, h5rewrite, h5writeattr, h5readattr
+
+### PUBLIC API ### (requires module scoping)
+
+### Public Functions
+# create_attribute, open_attribute. delete_attribute, read_attribute, write_attribute, attributes,
+# create_dataset, create_external_dataset, open_dataset, read_dataset, write_dataset,
+# create_group, open_group,
+# copy_object, open_object, delete_object,
+# create_datatype, commit_datatype,
+# create_property
+# group_info, object_info
+
+# file, filename, name,
+# dataspace, datatype,
```

Comment:
I'm open to exports lines 18-27. Although, personally I feel like we should keep the list of exports minimal.

Reply:
I think I'd tend toward exporting those. A lot of the high-level interface is already accessible with only `import HDF5` because it's implemented as overloads on many `Base` functions, so `using` vs `import` alone is already pretty good at cleaning up a namespace if necessary. I think the `using` case can afford to have a few more exports.